src/components/utils/generateURI.js

```diff
@@ -0,0 +1,13 @@
+function generateApiUri(baseUrl, endpoint, queryParams) {
+  const apiUri = new URL(endpoint, baseUrl);
+
+  if (queryParams) {
+    Object.keys(queryParams).forEach(key => {
+      apiUri.searchParams.append(key, queryParams[key]);
+    });
```

Comment:
- `P1`
해당 구문은
```suggestion
    for (const key in queryParams) {
      if (queryParams.hasOwnProperty(key)) {
        apiUri.searchParams.append(key, queryParams[key]);
      }
    }
```

이렇게 `iteration`을 줄일 수 있지 않을까요?

Reply:
`Object.keys`는 O(n) 인 반면에 `hasOwnProperty()`는 O(1)의 시간 복잡도를 갖고있네요
2n -> n 으로 시간복잡도를 줄일수 있어 좋겠군요 바로 반영하겠습니다.

Reply:
수정 반영이 안됬습니다!